src/i3ipc.ml

```diff
@@ -426,7 +426,10 @@ type connection = {
 
 let connect () =
   try%lwt
-    let%lwt socketpath = Lwt_process.pread ("i3", [|"i3"; "--get-socketpath"|]) in
+    let%lwt socketpath = match Sys.getenv_opt "I3SOCK" with
```

Comment:
Mmh, `Sys.getenv_opt` was introduced in ocaml 4.05, but at the moment the library supports ocaml >= 4.03.
Here you can quite easily rewrite the code to use `Sys.getenv` and the match-with-exception feature:
`match Sys.getenv "I3SOCK" with exception Not_found -> ... | s -> ...`

Reply:
Thanks for the review! I'll fix this